django_mongodb/base.py

```diff
@@ -157,20 +159,35 @@ def __getattr__(self, attr):
         raise AttributeError(attr)
 
     def _connect(self):
-        settings_dict = self.settings_dict
+        settings_dict = deepcopy(self.settings_dict)
+        user = settings_dict.get("USER")
+        password = settings_dict.get("PASSWORD")
+        options = settings_dict.get("OPTIONS", {})
+        options_user = options.pop("username", None)
```

Comment:
I think we should ignore the possibility the developer put username and password in OPTIONS as that's just creating extra complexity, and it breaks the assumption in other places that the USER/PASSWORD will be set (e.g. https://github.com/mongodb-labs/django-mongodb/blob/main/django_mongodb/client.py).